File: src/test/java/com/fractalwrench/bpp/ByteArrayClassLoaderTest.java
package com.fractalwrench.bpp;

import com.fractalwrench.bpp.internal.ast.RootNode;
import org.junit.Before;
import org.junit.Test;

import java.io.File;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class ByteArrayClassLoaderTest {

    private static final String CLASS_NAME = "Hello";
    private ByteArrayClassLoader byteArrayClassLoader;

    @Before
    public void setUp() throws Exception {
        byteArrayClassLoader = new ByteArrayClassLoader();
    }

    @Test
    public void loadClassFromByteCode() throws Exception {
        byte[] dump = RootNode.dump(CLASS_NAME);
        Class<?> hello = byteArrayClassLoader.loadClassFromByteCode(dump, CLASS_NAME);
        assertNotNull(hello);
        assertEquals(hello.getName(), CLASS_NAME);
    }

    @Test(expected = RuntimeException.class)
    public void loadClassFromFile() throws Exception {
        byteArrayClassLoader.loadClassFromFile(new File("fake.txt"), CLASS_NAME);
    }
}

File: src/test/java/com/fractalwrench/bpp/ByteArrayClassLoaderTest.kt
package com.fractalwrench.bpp

import com.fractalwrench.bpp.internal.ast.RootNode
import org.junit.Before
import org.junit.Test

import java.io.File

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull

class ByteArrayClassLoaderTest {
    private var byteArrayClassLoader: ByteArrayClassLoader? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        byteArrayClassLoader = ByteArrayClassLoader()
    }

    @Test
    @Throws(Exception::class)
    fun loadClassFromByteCode() {
        val dump = RootNode(null, null).generateClass(CLASS_NAME)
        val hello = byteArrayClassLoader!!.loadClassFromByteCode(dump, CLASS_NAME)
        assertNotNull(hello)
        assertEquals(hello.name, CLASS_NAME)
    }

    @Test(expected = RuntimeException::class)
    @Throws(Exception::class)
    fun loadClassFromFile() {
        byteArrayClassLoader!!.loadClassFromFile(File("fake.txt"), CLASS_NAME)
    }

    companion object {

        private val CLASS_NAME = "Hello"
    }
}